invokeai/app/services/graph.py

```diff
@@ -1114,9 +1128,22 @@ def _get_iteration_node(
         )
 
     def _get_next_node(self) -> Optional[BaseInvocation]:
+        """Gets the deepest node that is ready to be executed"""
         g = self.execution_graph.nx_graph()
-        sorted_nodes = nx.topological_sort(g)
-        next_node = next((n for n in sorted_nodes if n not in self.executed), None)
+
+        # we need to traverse the graph in from bottom up
+        reversed_sorted_nodes = reversed(list(nx.topological_sort(g)))
```

Comment:
I don't think you want a reversed topological sort. That won't work well if you queue up multiple nodes. You want a depth first topological sort... (Which might not be an accurate way of describing it, the depth first part is important).

Reply:
I can't say I understand fully why it won't work well - but depth-first search using the pre-order traversal strategy is a depth-first topological sort. Changing to that (it works as expected).